src/error.c

```diff
@@ -253,7 +254,7 @@ Obj FuncCALL_WITH_CATCH(Obj self, Obj func, volatile Obj args)
         CHANGED_BAG(res);
         STATE(ThrownObject) = 0;
         SWITCH_TO_OLD_LVARS(currLVars);
-        STATE(CurrStat) = currStat;
+        GAP_ASSERT(currStat == BRK_CALL_TO());
```

Comment:
This is a paranoia check, to verify I did not break anything fundamental in the refactoring. It could/should likely be removed eventually.

Reply:
I personally don't think you can have too many asserting paranoia checks :)